### PR TITLE
Bugfix/myilmaz vue preset bug

### PR DESCRIPTION
**Pull request by mylmz10**
api.resolve added to paths for project creation with vue --preset args
npm packages versions fixed
vuexp-router prompt added

### CODE REVIEW

File: generator/index.js [hunk 1]
```diff
@@ -6,7 +6,10 @@ const replace = require('replace-in-file');
 
 const newline = process.platform === 'win32' ? '\r\n' : '\n';
 
+const PluginVersion = require('../package.json').version;
+
 module.exports = async (api, options, rootOptions) => {
+  console.log(rootOptions);
```
Comment (brendtumi) [inline generator/index.js]: Lose unnecessary console

File: generator/index.js [hunk 2]
```diff
@@ -78,18 +81,18 @@ module.exports = async (api, options, rootOptions) => {
...
     devDependencies: {
-      'nativescript-dev-webpack': '~1.1.0',
-      'nativescript-vue-template-compiler': '~2.4.0',
-      'nativescript-worker-loader': '~0.9.5',
-      'node-sass': '^4.11.0',
-      'string-replace-loader': '^2.1.1',
-      rimraf: '^2.6.3',
-      'sass-loader': '^7.1.0'
+      'nativescript-dev-webpack': '1.1.0',
+      'nativescript-vue-template-compiler': '2.2.2',
```
Comment (brendtumi) [inline generator/index.js:89]: do we have a problem with 2.4.0 of nativescript-vue-template-compiler?

File: generator/index.js [hunk 3]
```diff
@@ -137,16 +132,18 @@ module.exports = async (api, options, rootOptions) => {
...
+        '@babel/preset-env': '7.4.5',
+        'babel-loader': '8.0.6',
+        '@babel/traverse': '7.4.5'
       }
     });
 
     api.render(async () => {
-      fs.ensureFileSync(genConfig.dirPathPrefix + 'babel.config.js');
-      await applyBabelConfig(api, genConfig.dirPathPrefix + 'babel.config.js');
+      console.log(api.resolve(path.join(genConfig.dirPathPrefix, 'babel.config.js')));
```
Comment (brendtumi) [inline generator/index.js]: Please remove console.log.

File: generator/index.js [hunk 4]
```diff
@@ -281,15 +287,21 @@ module.exports = async (api, options, rootOptions) => {
...
-const vueRouterSetup = (module.exports.vueRouterSetup = async (api, filePathPrefix, jsOrTs) => {
+const routerSetup = (module.exports.routerSetup = async (api, rootOptions, filePathPrefix, jsOrTs, vuexpRouter) => {
   try {
-    if (api.hasPlugin('vue-router')) {
-      api.injectImports(filePathPrefix.replace(/.\//, '') + 'src/main' + jsOrTs, `import router from './router';`);
-      api.injectRootOptions(filePathPrefix.replace(/.\//, '') + 'src/main' + jsOrTs, `router`);
+    if (vuexpRouter) {
+      api.extendPackage({
+        dependencies: {
+          'vuexp-router': 'git://github.com/vuexp/vuexp-router.git#development'
```
Comment (brendtumi) [inline generator/index.js:295]: is there any reason we getting latest version of development branch?

File: generator/index.js [hunk 5]
```diff
@@ -347,14 +357,15 @@ const applyBabelConfig = (module.exports.applyBabelConfig = async (api, filePath
 // Typescript projects.  for new projects it will write files to the root of the project
 // and for existing projects it will write it to the ./vuexp-example directory
 const writeRootFiles = (module.exports.writeRootFiles = async (api, options, filePathPrefix) => {
+  console.log('writeRootFiles');
```
Comment (brendtumi) [inline generator/index.js]: why eslint let you pass those console.logs ?

File: generator/index.js [hunk 6]
```diff
@@ -478,9 +489,11 @@ const gitignoreAdditions = (module.exports.gitignoreAdditions = async (api) => {
 
 // setup nsconfig.json file.  for new projects it will write to the root of the project
 // and for existing projects it will write it to the ./vuexp-example directory
-const nsconfigSetup = (module.exports.nsconfigSetup = async (dirPathPrefix, nsconfigPath, nativeAppPathModifier, appResourcesPathModifier, options) => {
+const nsconfigSetup = (module.exports.nsconfigSetup = async (api, dirPathPrefix, nsconfigPath, nativeAppPathModifier, appResourcesPathModifier, options) => {
   let nsconfigContent = '';
 
+  console.log('nsconfigPath', nsconfigPath);
```
Comment (brendtumi) [inline generator/index.js]: c'mon :)

**Review by osmanertem — DISMISSED** [review]

**Review by brendtumi — APPROVED** [review]

**Pull request merged**